compiler/passes/convert-typed-uast.cpp

```diff
@@ -5071,6 +5090,157 @@ bool TConverter::enter(const ExternBlock* node, RV& rv) {
 
 void TConverter::exit(const ExternBlock* node, RV& rv) {}
 
+static SymExpr* makeCaseCond(TConverter& tc, TConverter::RV& rv,
+                             const uast::When* when,
+                             SymExpr* selectExpr,
+                             const uast::AstNode* cs) {
+  auto re = rv.byAst(cs);
+
+  // Grab the '==' ResolvedFunction
+  const AssociatedAction* action = nullptr;
+  for (const auto& a : re.associatedActions()) {
+    if (a.action() == AssociatedAction::COMPARE) {
+      action = &a;
+      break;
+    } else {
+      INT_ASSERT(false);
+    }
+  }
+  auto cmp = action->fn();
+
+  // TODO: create a wrapper for this kind of thing
+  const ResolvedFunction* rf = nullptr;
+  if (tc.paramElideCallOrNull(cmp, re.poiScope(), &rf)) {
+    INT_ASSERT(false && "Should not have param elided initializer call!");
+  } else if (!rf) {
+    return toSymExpr(TC_PLACEHOLDER((&tc)));
+  }
+
+  auto fn = tc.findOrConvertFunction(rf);
+  INT_ASSERT(fn);
+
+  // TODO: is there a better way to represent the return type of '=='? As-is,
+  //   it's a bit confusing for the case-expression's type in 'rv' to always be
+  //   a bool.
+  Expr* caseExpr = tc.convertExpr(cs, rv);
+  caseExpr = tc.storeInTempIfNeeded(caseExpr, {});
```

Comment:
Can we get the type from `convertExpr` and pass that to `storeInTempIfNeeded` here?